src/deployment/deployment.ts

```diff
@@ -150,12 +150,6 @@ export class Deployment {
 
     await this.dataTransferProtocol.init();
 
-    // TODO: allocation is not used in this api?
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const allocation = await this.modules.payment.createAllocation({
-      budget: 1, // TODO: Shouldn't this be generated by the deployment internally when the demand is known?
-      expirationSec: 24 * 60 * 60,
-    });
```

Comment:
Yeah, it's not used now because agreement pool doesn't take an allocation, but in the parallel universe where we have `leaseProcessPool` it uses this allocation, please leave it here and if possible remove the hard coded budget and expirationSec

Reply:
Addressing this type of allocation within the context of the deployment isn't going to be fast. I'd like to discuss my findings from the implementation attempt I just made, so that we can design how this should look in it's ultimate form - instead of a hack.